pkg/apis/pipeline/v1alpha1/task_validation.go

```diff
@@ -173,6 +213,47 @@ func validateResourceVariables(steps []corev1.Container, inputs *Inputs, outputs
 	return validateVariables(steps, "resources", resourceNames)
 }
 
+func validateArrayUsage(steps []corev1.Container, prefix string, vars map[string]struct{}) *apis.FieldError {
```

Comment:
Woah, I didn't think about this :) Is this a complete list of all the step fields that we need to check?
Hopefully if we ever change the spec of Steps, we'll remember to update the validation here too.

Reply:
I replied to this 3 days ago and then eventually resolved it, but apparently my reply was either wiped or never sent :angry: 

I believe that is a complete list of all step fields that need to be checked; the step field checking logic is directly copied from `validateVariables()`. 

Something to consider is that in the future, if the spec is changed, someone might remember to update `validateVariables()` but forget to add that extra field to `validateArrayUsage()`. I don't believe those 2 methods can be combined, though, because `validateArrayUsage()` does distinctly different checks on each field in comparison to `validateVariables()`, like `validateTaskNoArrayReferenced` and `validateTaskArraysIsolated`. Additionally, `validateVariables()` is called in different contexts that wouldn't make sense to use array validation logic for (like resource validation).

Reply:
I just realized that each method is checking the fields of `corev1.container`, not a type we define ourselves, so I don't think it's as necessary to add an reminder comment. Let me know if you disagree.